rig/machine_control/machine_controller.py

```diff
@@ -2110,6 +2132,19 @@ def chips(self):
         """
         return iter(self)
 
+    def ethernet_connected_chips(self):
+        """Iterate over the coordinates of Ethernet connected chips.
+
+        Yields
+        ------
+        ((x, y), str)
+            The coordinate and IP address of each Ethernet connected chip in
+            the system.
+        """
+        return ((xy, chip_info.ip_address)
```

Comment:
I'm not really a fan of returning the generator, sorry.

I think

```python
for xy, chip_info in six.iteritems(self):
    if chip_info.ethernet_up:
       yield xy, chip_info
```

would be clearer